desktop/core/src/desktop/js/reactComponents/FileChooser/PathBrowser/PathBrowser.tsx

```diff
@@ -0,0 +1,192 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useRef, useEffect, useState, RefObject } from 'react';
+import { Input, Button, Dropdown } from 'antd';
+import type { MenuProps } from 'antd';
+import { RightOutlined } from '@ant-design/icons';
+
+import HdfsIcon from '../../../components/icons/HdfsIcon';
+import S3Icon from '../../../components/icons/S3Icon';
+import AdlsIcon from '../../../components/icons/AdlsIcon';
+
+import { BreadcrumbData } from '../types';
+import OverflowingItem from './OverflowingItem';
+import './PathBrowser.scss';
+
+interface PathBrowserProps {
+  breadcrumbs?: BreadcrumbData[];
+  handleFilePathChange: (path: string) => void;
+}
+
+const PathBrowser: React.FC<PathBrowserProps> = ({ breadcrumbs, handleFilePathChange }) => {
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  const icons = {
+    //hdfs file system begins with the first breadcrumb as "/" (ex: /user/demo)
+    '/': <HdfsIcon />,
+    abfs: <AdlsIcon />,
+    s3: <S3Icon />
+  };
+
+  const useOutsideAlerter = (ref: RefObject<HTMLDivElement>) => {
+    useEffect(() => {
+      // Alert if clicked on outside of element
+      const handleClickOutside = (event: MouseEvent) => {
+        const current = ref?.current;
+        if (current && !current.contains(event.target as Node)) {
+          setIsEditMode(false);
+        }
+      };
+      // Bind the event listener
+      document.addEventListener('mousedown', handleClickOutside);
+      return () => {
+        // Unbind the event listener on clean up
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    }, []);
+  };
+
+  const extractFileSystem = (label: string) => {
+    const prefix = label.substring(0, label.length - 3);
```

Comment:
can you rename the prefix variable to be more descriptive, something like **fileSystem** or any other alternative